### PR TITLE
Native support for styled.{{tag}}

### PR DESCRIPTION
**Pull request by zakybilfagih**
When trying to use `styled.{{tag}}` on native, got an error `Uninterpreted extension 'raw'`
This is because `server-reason-react.melange_ppx` doesn't recognize the `raw` extension

### CODE REVIEW

**Comment by davesnx** [comment]
That's actually a nice improvement, since it fixes the mel.raw issue but you will face other issues with styled.tag in native (mel.raw, bindings such as createVariadicElement).

I would propose to add a flag on the ppx (styled-ppx.lib --native) and have a different creation of elements given the flag.

**Comment by zakybilfagih** [comment]
I see.. I guess the first step is to add `server-reason-react.browser_ppx` library to `styled-ppx.lib`, and set it to add the --js flag when --native flag is not set?

**Comment by zakybilfagih** [comment]
> I see.. I guess the first step is to add `server-reason-react.browser_ppx` library to `styled-ppx.lib`, and set it to add the --js flag when --native flag is not set?
aah nvm, this should be handled by the user (?) we just need to emit valid native code right?

**Comment by zakybilfagih** [comment]
I felt the build time is quite slow (tried building `e2e/melange/native/ui.re` and it felt considerably slower than the non native variant with `styled.{{tag}}`). Can't seem to pin point the slow path.

Is using a lot of ppxlib.metaquot have an impact on build times? It shouldn't affect the build time for the user of the ppx right?

**Comment by davesnx** [comment]
That should have any effect, while building ui (maybe we copy the node_modules by accident?)

File: packages/ppx/test/snapshot/rescript/rescript-static.t/run.t [hunk 1]
```diff
@@ -105,7 +105,7 @@
         contextMenu: string [@bs.optional ];
         controls: bool [@bs.optional ];
         coords: string [@bs.optional ];
-        crossorigin: bool [@bs.optional ];
+        crossOrigin: string [@bs.optional ];
```
Comment (davesnx) [inline packages/ppx/test/snapshot/rescript/rescript-static.t/run.t:108]: Crazy good fix

File: packages/ppx/test/snapshot/rescript/rescript-static.t/run.t [hunk 2]
```diff
@@ -235,7 +235,7 @@
         maxLength: int [@bs.optional ];
         media: string [@bs.optional ];
         mediaGroup: string [@bs.optional ];
-        min: int [@bs.optional ];
+        min: string [@bs.optional ];
```
Comment (davesnx) [inline packages/ppx/test/snapshot/rescript/rescript-static.t/run.t:238]: Same

**Review by davesnx — APPROVED** [review]
Looks amazing. Well done

File: packages/ppx/test/snapshot/rescript/rescript-static.t/run.t (hunk 1 above)
Comment (zakybilfagih) [reply packages/ppx/test/snapshot/rescript/rescript-static.t/run.t:108]: Calling `React.domProps` caught this 😆

**Comment by zakybilfagih** [comment]
> That should have any effect, while building ui (maybe we copy the node_modules by accident?)

That doesn't seem to be the case, I tried timing the build command. Just adding the styled.{{tag}} call, cause 2x the build
watzz
firz

**Comment by davesnx** [comment]
Maybe the amount of code generated? Can you debug without any prop?

**Comment by zakybilfagih** [comment]
welp, having a dozen of arguments (ReactDOM.domProps) will inevitably effect the build time. nothing we can do about it

**Pull request merged**